### ISSUE TITLE
New collection discarded when aborting also pending media sync

### ISSUE THREAD

**Issue by nwwt** [ankidroid/Anki-Android#7819]
###### Reproduction Steps

1. On another device, do a full sync + enough media file changes.
2. On AnkiDroid, start a full sync to pull these.
3. Finish the collection download/profile sync but abort the ongoing media sync.

###### Expected Result

AnkiDroid applies the downloaded collection/database.

###### Actual Result

AnkiDroid discards it.

(Inconvenient since it thus needs to redundantly be redownload when resuming the media sync and one can't proceed reviewing on the device in the meantime, either.

Noticed since atm media downloads are incredibly slow: As if there was a ~12s idle after each 25 files batch. Meanwhile the database download is full speed as ever.)

###### Debug info
2.1.5a9

**Comment by mikehardy** [comment]
Hi there!
I too have noticed the media sync pause between batches. I think that's a separate issue and could maybe be logged as such - could be that fixing that is sufficient to make media sync more convenient, then relying on it to finish isn't such a big deal?

And media sync slowness might just be a connection reuse issue which we have fixed before, I can look

I think with conversion to Rust this becomes a different issue though as media sync can move to background and be done separately - basically the whole sync infra can move to Rust and then we'll have new+different problems but bugs on the current Java sync code may be irrelevant.

**Comment by nwwt** [comment]
Yes, it would be, I've got a couple of thousand new files and watching it drag on is… painful. Ofc, if the matter hopefully becomes redundant with the switch to Rust, then I can wait for it to finish and use the other device in the meantime.

Incidentally, I found a "workaround": Apparently (wasn't looking at the device at the time) if the media sync bugs out on its own, then the new collection does get applied. Something about no longer being able to resolve/reach the ankiweb address (guess it's sleep related, as my wifi is stable). So manually killing the connection would probably have sufficed.
